src/platform/plugins/shared/dashboard/public/dashboard_top_nav/internal_dashboard_top_nav.tsx

```diff
@@ -152,6 +155,9 @@ export function InternalDashboardTopNav({
             <>
               {dashboardTitle}
               <EuiIcon
+                tabIndex={0}
```

Comment:
slightly fixing a11y for this edit button, but I wonder if we should improve it further and use the same placement pattern and a real button (EuiButtonIcon instead of EuiIcon) for it